### ISSUE TITLE
Maximum call stack size exceeded on Mobile chrome

### ISSUE THREAD

**Issue by moore** [elm-lang/elm-lang.org#671]
The simple app in this repo:

  https://github.com/alicious/warpengine

Works on desktop chrome but fails on mobile chrome with the error: 

  index.html:1 Uncaught RangeError: Maximum call stack size exceeded

This is happening at load time before the app has been loaded with : Elm.Warp.fullscreen() 

This is in elm 0.18.

**Comment by moore** [comment]
I think we are getting close to understanding this bug. It seems connected to having lists with hundreds of elements in them. None of our code is recursive but I am guessing some of elm's List functions are. Or that the list construction is treated by the browser as recursive calls.

**Comment by moore** [comment]
I have reduced this bug to loading a file with a large literal list such as:
 https://github.com/alicious/warpengine/blob/master/DoubleSide.elm

With a larger list, the same error on the desktop browser too.

The fundamental issue is that Elm represents lists as linked lists, that there is only an List and no Array literal, and constructing the Lists at js compile time causes the stack depth to be exceeded.

I have worked around this by storing a string "[1,2,3...n]" and using a json decoder to decode to array. This works but is unsatisfactory and points to a weakness of Elm in programs with large constant data sets.  

Also it is a run time error. At the least, the compiler should complain about these large lists.
